src/Common/src/System/Net/Logging/GlobalLog.cs

```diff
@@ -253,13 +253,10 @@ public static void Dump(byte[] buffer, int offset, int length)
                 return;
             }
 
-            var bufferSegment = new byte[length];
-            for (int i = 0; i < length; i++)
-            {
-                bufferSegment[i] = buffer[offset + i];
-            }
+            ArraySegment<byte> bufferSegment = new ArraySegment<byte>(buffer, offset, length);
+            byte[] bufferSegmentArray = bufferSegment.Array;
```

Comment:
You're right Pat, then that's probably the reason it was changed in the port.

I'm going to think another solution.